scripts/smoke-published.ts

```diff
@@ -0,0 +1,369 @@
+  packageManager: PackageManager;
+}) {
+  const turboBinary = getTurboBinary({ installType, packageManager });
+  exec({
+    title: "Verify second turbo build is successful and cached",
+    command: `${turboBinary} build`,
+    conditions: [
+      { expected: "2 successful, 2 total", operator: "includes" },
+      { expected: "2 cached, 2 total", operator: "includes" },
+      { expected: "FULL TURBO", operator: "includes" },
+    ],
+  });
+}
+
+function local({ local, packageManager }: TestInput) {
+  // setup example
+  installExample({ version: local.version, packageManager });
+  verifyLocalBinary({ installType: "local", packageManager });
+  logTurboDetails({ installType: "local", packageManager });
+
+  // verify build is correctly cached
+  verifyFirstBuild({ installType: "local", packageManager });
+  verifySecondBuild({ installType: "local", packageManager });
+}
+
+function global({ local, global, packageManager }: TestInput) {
+  // setup example
+  installExample({ version: local.version, packageManager });
+  installGlobalTurbo({ version: global.version, packageManager });
+  logTurboDetails({ installType: "global", packageManager });
+
+  verifyLocalBinary({ installType: "global", packageManager });
+  uninstallLocalTurbo({ packageManager });
+  logTurboDetails({ installType: "global", packageManager });
+  verifyGlobalBinary({ installType: "global", packageManager });
+
+  // verify build is correctly cached
+  verifyFirstBuild({ installType: "global", packageManager });
+  verifySecondBuild({ installType: "global", packageManager });
+}
+
+function both({ local, global, packageManager }: TestInput) {
+  // setup example
+  installExample({ version: local.version, packageManager });
+  installGlobalTurbo({ version: global.version, packageManager });
+  logTurboDetails({ installType: "global", packageManager });
+  verifyLocalBinary({ installType: "global", packageManager });
+
+  // verify build is correctly cached
+  verifyFirstBuild({ installType: "global", packageManager });
+  verifySecondBuild({ installType: "global", packageManager });
+}
+
+const tests = {
+  local,
+  global,
+  both,
+};
+
+/**
+ * Triggered via github workflow manually, or on publish of a new version
+ *
+ * https://github.com/vercel/turbo/blob/main/.github/workflows/turborepo-smoke-published.yml
+ */
+function test() {
+  const args = process.argv.slice(2) as Args;
+  const [
+    testName = "local",
+    packageManager = "pnpm",
+    localVersion = "canary",
+    globalVersion = "canary",
+  ] = args;
+
+  const local: VersionInput = {
+    type: "local",
+    version: localVersion,
+  };
+  const global: VersionInput = {
+    type: "global",
+    version: globalVersion,
+  };
+
+  console.log(
+    `Running test: "${testName}" with local version: "turbo@${localVersion}" and global version: turbo@${globalVersion} using ${packageManager}`
+  );
+  tests[testName]({ local, global, packageManager });
+  console.log("Tests passed!");
+}
+
+test();
```

Comment:
Does it make sense to check if this is the main module before running this? Might be useful in case others want to use this code.